packages/component-base/src/data-provider-controller/data-provider-controller.js

```diff
@@ -140,16 +149,36 @@ export class DataProviderController extends EventTarget {
 
   /**
    * Returns context for the given flattened index, including:
-   * - the corresponding cache
-   * - the associated item (if loaded)
-   * - the corresponding index in the cache's items array.
-   * - the page containing the index.
-   * - the cache level
+   * - the corresponding cache.
+   * - the cache level.
+   * - the corresponding item (if loaded).
+   * - the item's index in the cache's items array.
+   * - the page containing the item.
+   *
+   * @param {number} flatIndex
    */
   getFlatIndexContext(flatIndex) {
     return getFlatIndexContext(this.rootCache, flatIndex);
   }
 
+  /**
+   * Returns context for the given item id, including
+   * - the cache containing the item.
+   * - the cache level.
+   * - the item.
+   * - the item's index in the cache's items array.
+   * - the item's flattened index.
+   * - the item's sub-cache (if exists).
+   * - the page containing the item.
+   *
+   * If no item with the given id is found, the method returns undefined.
+   *
+   * @param {unknown} itemId
+   */
+  getItemContext(itemId) {
```

Comment:
The reason I initially preferred `itemId` to `item` is because the grid connector often doesn't have the whole item object but its id. However, now I tend to think it was the wrong choice. The `itemId` terminology can be confusing when the grid is filled with primitive items (not objects).

```suggestion
  getItemContext(item) {
```

Reply:
Updated.